engine/app/helpers/good_job/application_helper.rb

```diff
@@ -1,24 +1,54 @@
 # frozen_string_literal: true
 module GoodJob
   module ApplicationHelper
-    def relative_time(timestamp)
-      text = timestamp.future? ? "in #{time_ago_in_words(timestamp)}" : "#{time_ago_in_words(timestamp)} ago"
+    def format_duration(sec)
+      return unless sec
+
+      if sec < 1
+        format "%dms", sec * 1000
+      elsif sec < 10
+        format "%.2fs", sec
+      elsif sec < 60
+        format "%ds", sec
+      elsif sec < 3600
+        format "%dm%ds", sec / 60, sec % 60
+      else
+        format "%dh%dm", sec / 3600, (sec % 3600) / 60
+      end
+    end
+
+    def relative_time(timestamp, **args)
+      text = timestamp.future? ? "in #{time_ago_in_words(timestamp, **args)}" : "#{time_ago_in_words(timestamp, **args)} ago"
       tag.time(text, datetime: timestamp, title: timestamp)
     end
 
+    STATUS_ICONS = {
+      discarded: "exclamation",
+      finished: "check",
+      queued: "clock",
```

Comment:
What do you think of this icon for queued? https://icons.getbootstrap.com/icons/list-nested/

Reply:
Another option would be to change "Scheduled" to be something like this [calendar event ](https://icons.getbootstrap.com/icons/calendar-event/) and then use the Clock icon for queued. 

This is like an advanced naming problem 🤣

Reply:
Hmmm. The nested-list looks a little odd because everything else is in a circle:

<img width="112" alt="Screen Shot 2022-05-17 at 8 26 22 AM" src="https://user-images.githubusercontent.com/47554/168850402-95ba8c1c-82d9-4a5c-9eca-f95b7df11ce4.png">

I think I'm going with dash-circle. I also changed the color to warning (which is waaaay to bright, but we can color them later:

<img width="109" alt="Screen Shot 2022-05-17 at 8 32 43 AM" src="https://user-images.githubusercontent.com/47554/168850610-c621d8be-521b-43d9-9c3c-7af20e93958c.png">